content/docs/source-git/how-to-source-git.md

```diff
@@ -1,218 +1,182 @@
 
-We also (currently) have to remove them from the spec file.
-Packit will automatically generate patches from the additional commits and
-add them to the spec when generating a SRPM or proposing downstream update.
-
-We can now commit the files in `fedora/` directory:
-```bash
-$ git commit -m 'add fedora packaging'
-[2437-sg 578bda8d7d] add fedora packaging
- 15 files changed, 3077 insertions(+)
- create mode 100644 fedora/.gitignore
- create mode 100755 fedora/20-grubby.install
- create mode 100644 fedora/20-yama-ptrace.conf
- create mode 100644 fedora/inittab
- create mode 100755 fedora/purge-nobody-user
- create mode 100644 fedora/sources
- create mode 100644 fedora/split-files.py
- create mode 100644 fedora/sysctl.conf.README
- create mode 100644 fedora/systemd-journal-gatewayd.xml
- create mode 100644 fedora/systemd-journal-remote.xml
- create mode 100644 fedora/systemd-udev-trigger-no-reload.conf
- create mode 100644 fedora/systemd-user
- create mode 100644 fedora/systemd.spec
- create mode 100644 fedora/triggers.systemd
- create mode 100644 fedora/yum-protect-systemd.conf
+Let's commit our changes:
+```
+$ cd ../
+$ git add .
+$ git commit -m "Fedora downstream packaging"
 ```
 
-
-### Applying downstream patches
-We are getting to the core of source-git: we work with code in it, not with
-patches. But first we need to instruct packit from which point in history start
-downstream patches.
-Add [upstream_ref](https://packit.dev/docs/configuration/#upstream-ref)
-key into `.packit.yaml`:
+How does our git history look now?
 
 ```
-upstream_ref: 2437-sg-start
+$ git log --oneline HEAD^^..
+3d2f796 (HEAD -> fedora-rawhide) Fedora downstream packaging
+d327cfe (tag: 4.0, origin/master, origin/HEAD, master) nts: save new server keys on start
 ```
 
-Commit and tag it:
+### Downstream patches
 
-```bash
-$ git commit -am "upstream_ref -> .packit.yaml"
+It's a common practice to pick up fixes from the master branch of an upstream
+project and backport them to the current (older) version of a package in a
+distribution. Alternatively to accommodate changes specific to the distribution
+which are not merged upstream.
 
-$ git tag 2437-sg-start HEAD
-```
-
-Now we can apply the downstream patches:
+In this stage, we are going to apply all downstream patches we have defined in
+our spec file.
 
-```bash
-$ git am fedora/464a73411c13596a130a7a8f0ac00ca728e5f69e.patch
-Applying: udev: use bfq as the default scheduler
+Right now there is only a single patch chrony has in rawhide, let's apply it:
+```
+$ curl -O https://src.fedoraproject.org/rpms/chrony/raw/master/f/chrony-nm-dispatcher-dhcp.patch
+$ git am chrony-nm-dispatcher-dhcp.patch
+Applying: examples/nm-dispatcher.dhcp: use sysconfig, detect dhclient
+```
 
-$ git am fedora/0002-Revert-units-set-NoNewPrivileges-for-all-long-runnin.patch
-Applying: Revert "units: set NoNewPrivileges= for all long-running services"
+The problem we have right now is that:
+1. the patch is defined in the spec file
```

Comment:
I have a problem with the word "defined", I'd use
```suggestion
1. the patch is applied in the spec file
```

Reply:
yes, this sounds like a better wording from the PoV of package maintainers - for me it's still defined but I can understand this is not the word other people would expect here